### ISSUE TITLE
[1.4.4] Item glitch in multiplayer

### ISSUE THREAD

**Issue by JustMrPhoenix** [blushiemagic/MagicStorage#254]
While playing on a server with one other player I have encountered the following glitch:

- They used a quick stack feature near the storage heart
- Upon opening the storage they saw that an item they had was stacked and the amount was raised to 214. The item in question was a [Piggy Bank](https://terraria.fandom.com/wiki/Piggy_Bank)
- The storage now had 214 piggy banks which they took back out
- Later upon opening the interface I noticed an empty spot after the first item
- I have realized something went wrong and have shut-down the server to prevent further damage
- Upon expecting the world in single-player there there was another item in the storage that most certainly didn't belong there [Meteorite Bar](https://terraria.fandom.com/wiki/Meteorite_Bar) in the same  quantity (214)

I later found three exceptions in the server log (2 at the moment of the bug). Server log and world backup attached
[server.log](https://github.com/blushiemagic/MagicStorage/files/12255944/server.log)
[2023-08-04-Uncombative_Eater.zip](https://github.com/blushiemagic/MagicStorage/files/12255945/2023-08-04-Uncombative_Eater.zip)

**Comment by sxyugao** [comment]
I have encoutered the same problem while playing with other players.

<img width="439" alt="ba773e8b48463762179182cd5a7c6235" src="https://github.com/blushiemagic/MagicStorage/assets/40171492/b1eeee83-25a2-4b2e-baec-c9237cb474f6">

![image](https://github.com/blushiemagic/MagicStorage/assets/40171492/585d0d75-f49e-4907-b697-74778fc311ed)

World file：

[Calamity.zip](https://github.com/blushiemagic/MagicStorage/files/12262741/Calamity.zip)

**Comment by JustMrPhoenix** [comment]
I got a comment on the issue on discord. Going to post it here too:

TLDR: Fix in the next version

![image](https://github.com/blushiemagic/MagicStorage/assets/14361083/e2e8dc99-3ce4-4d19-ae12-266648dad4cc)

Going to keep the issue open until I can test it myself and see it fixed

**Comment by JustMrPhoenix** [comment]
Was fixed in v0.6.0.2
